### ISSUE TITLE
Gulp watch race condition

### ISSUE THREAD

**Issue by reneolivo** [jakemmarsh/angularjs-gulp-browserify-boilerplate#108]
When you update a stylesheet and the operation takes one or two seconds the browser reloads before the stylesheet is processed. This causes the browser to reload with the previous changes.

I'm guessing the problem applies to all other watch sequences, but I have only experienced it with the styles.

**Comment by jakemmarsh** [comment]
Looks like it's recommended to create a separate task for reloading browser-sync: https://github.com/BrowserSync/browser-sync/issues/535

**Comment by reneolivo** [comment]
That didn't help me at all anyway because gulp tasks run in parallel, not in sequence (at least not by default). 

I have been doing several tests and I think I found the culprit: `browserSync.stream({once: true});`. More specifically `{once: true}`.

I'm not sure if I understand it well (the documentation is not clear), but it seems that `{once: true}` is so that when all the tasks run they don't call the reload method a bunch of times, and I don't know what's going on internally, but that makes browserSync reload the browser before the task is completed. For example:

```
[20:31:12] Starting 'styles'...
[20:31:12] Starting 'reload-browser'...
[BS] Reloading Browsers...
[20:31:12] Finished 'reload-browser' after 1.01 ms
[20:31:16] Finished 'styles' after 3.3 s
```

No `{once: true}`:

```
[20:37:09] Starting 'styles'...
[BS] 1 file changed (main.css)
[20:37:12] Finished 'styles' after 3 s
```

Cool thing is: it doesn't even do a full reload!!! It just changed the CSS files on the fly.

Again: I think the `{once: true}` is there to stop piling up too many reloads at the same time, but I don't think removing it is going to be an issue (not 100% sure) since those tasks (styles, views, images, etc.) run before browserSync is started

**Comment by DrewML** [comment]
I think the fix suggested for this is correct.

[From the BrowserSync stream source](https://github.com/BrowserSync/browser-sync/blob/master/lib/public/stream.js#L52):

```javascript
            /**
             * if {once: true} provided, emit the reload event for the
             * first file only
             */
            if (opts.once === true && !emitted) {

                utils.emitBrowserReload(emitter);

                emitted = true;

            } else { // handle multiple

                if (opts.once === true && emitted) {

                } else {

                    if (file.path) {

                        emitted = true;
                        utils.emitChangeEvent(emitter, file.path, false);
                        changed.push(path.basename(file.path));
                    }
                }
            }
```

BrowserSync emits a full reload when `once` is true. Further, it does it on the first file processed, which could explain the race-case.

Also found [this note from the owner of the project](https://github.com/BrowserSync/browser-sync/issues/175#issuecomment-48837247), saying:

> having `once: true` will NEVER result in css injection`

**Comment by jakemmarsh** [comment]
Fixed in #109
